src/shared/featureController.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SettingsConfiguration } from './settingsConfiguration'
+
+// This enum represents the list of active feature flags to check for
+// Features all need to be added with matching enums and string values
+// e.g. `Feature1 = 'Feature1'`
+// Features that do not follow this scheme will not work.
+// You cannot have more active features than FeatureController.maxFeatures (default: 5)
+// Any features that are flagged in the code but not added here will always return false.
+export enum ActiveFeatureKeys {
+}
+
+/**
+ * This class handles feature access for unreleased or gated features.
+ * Example: adding a feature called `NewFeature1`:
+ *
+ * Add the following to the `featureController.ActiveFeatureKeys` enum:
+ *
+ * ```
+ * export enum ActiveFeatureKeys {
+ *     NewFeature1 = 'NewFeature1'
+ * }
+ * ```
+ *
+ * You can then gate your feature with the following code (returning a boolean):
+ *
+ * ```
+ * FeatureController.isFeatureActive(featureController.ActiveFeatureKeys.NewFeature1)
+ * ```
+ *
+ * Finally, you can access the gated feature by adding the following snippet to the VS Code `settings.json` file:
+ *
+ * ```
+ *     "aws.experimentalFeatureFlags": [
+ *         "NewFeature1",
+ *     ],
+ * ```
+ */
+export class FeatureController {
+
+    private readonly enabledFeatures: Set<string>
+
+    private readonly maxFeatures: number = 5
```

Comment:
Late comment, but what is the point of ```maxFeatures```? I don't imagine we'll have more than 5 features in flight at the same time, but what is the point of limiting it?

Reply:
I think I quote you when I say this, but it's to avoid https://news.ycombinator.com/item?id=18442941